src/array.rs

```diff
@@ -448,7 +448,7 @@ impl<T: Element, D: Dimension> PyArray<T, D> {
         ID: IntoDimension<Dim = D>,
     {
         let dims = dims.into_dimension();
-        let data_ptr = data_ptr.unwrap_or(boxed_slice.as_ptr());
+        let data_ptr = data_ptr.unwrap_or_else(|| boxed_slice.as_ptr());
```

Comment:
I should have pointed out when reviewing (I thought recent clippy is OK with this...)

Reply:
Exactly, current nightly Clippy actually wants the version without the closure whereas current stable Clippy complains about it. I think nightly Clippy is right, as `boxed_slice.as_ptr()` does not cost anything that should be delayed by wrapping it in a closure. So I expect to change this again with the next stable Clippy release. But as written above, the most important thing is to have a systematic check of this stuff via the CI.